contrib/gofiber/fiber.v2/fiber.go

```diff
@@ -60,7 +60,6 @@ func Middleware(opts ...Option) func(c *fiber.Ctx) error {
 		if spanctx, err := tracer.Extract(tracer.HTTPHeadersCarrier(h)); err == nil {
 			opts = append(opts, tracer.ChildOf(spanctx))
 		}
-		opts = append(opts, cfg.spanOpts...)
```

Comment:
This is actually ok because we're not modifying `cfg.spanOpts`, but the new code is fine too.

Reply:
But we're returning a function that does this.. I'm imagining this scenario:

1. someone defines some `opts`
2. someone called `Middleware` with those `opts`
3. `Middleware` creates a config based on those `opts` and uses it in a closure
4. they now have a `func(c *fiber.Ctx) error` that creates a span, using the `opts` passed by the user
5. [THE PROBLEM] A user changes the `opts` at this time. This could affect that returned function, right?

Reply:
> they now have a func(c *fiber.Ctx) error that creates a span, using the opts passed by the user

The `opts` argument itself is not the issue here. The `opts` inside the closure is a new variable with a different type, unrelated to the `opts` passed in by the user (created on line 40).

The `opts` passed in by the user only live as long as the `Middleware()` call. The only danger here is the `cfg.spanOpts` being settable by the user, so they cannot be modified. `opts = append(opts, cfg.spanOpts...)` doesn't modify the `spanOpts`, but correctly copies them into the local `opts`.

Reply:
We *could* be defensive and copy `spanOpts` so that users can't modify it after they call us, but we'd have to do that outside the closure before we return from `Middleware`.

Reply:
Discussed offline - we'll keep it as is because it was safe before, and this could change order which could have side effects. It's not consistent, but better to not possibly break people.